### PR TITLE
Vali Spear: The Return

### PR DESCRIPTION
**Pull request by Barnet2**
## About The Pull Request
Re-adds vali spear and it's sprites.
## Why It's Good For The Game
When vali spear was removed, I supported the PR. On retrospective the spear playstyle was mostly only enabled to be as annoying as it was due to the things around it - Russian Red, many xenos not being as strong as they are now and jetpack pre throw refactor.
The removal PR has one main point, it seems: "Vali spear is a mistake because it creates riskless gameplay where as vali is defined by risk." However, with spear, the reason they could get away with this "riskless gameplay" was not tied to the two tile range of the spear (that gap can be closed quite literally instantly) but to the insane healing you could stack ONTOP of Vali; thus Vali was an issue itself until the things above were touched/removed. Another issue presented is that Vali can kite (apart of the "riskless gameplay"), which is not exactly wrong compared to the other two weapons, as you can get hit easier when you try to return a hit with one tile range. I can only see that being a problem with Tramadol slow - if a maintainer requests that to be reduced/removed on spear I will do so.
:cl:
add: Vali spear has been re-added.
/:cl:

### CODE REVIEW

**Comment by Wisemonster** [comment]
Wasn't the reason it was removed due to the extra range allowing vali spears to just out range most non-spitter castes? Walls were also made weaker to melee, so vali spear would be better at maze clearing than it was before, due to not needing to hug walls to clear them.

**Comment by 74delta** [comment]
Vali spear was never really the problem either imo, the problem with vali has always been its healing cause if you really just want to win you'll get greenblood drop your weapon and use a gun. 

The slowdown effect from tram with the two tile poke while annoying was only ever a death sentence if you had no way to stun or cc. 

Stuns and cc didn't matter much at the time cause russian red combined with old nanites plus medium or heavy armor ontop of regular bktt + valis healing made you an unstoppable juggernaut that didn't care if you got stunned or not and could just relentlessly poke and prod a xeno to death. 


The game has changed alot to the point where I think vali spear deserves a 2nd chance if nothing else.

**Comment by Barnet2** [comment]
> Wasn't the reason it was removed due to the extra range allowing vali spears to just out range most non-spitter castes? Walls were also made weaker to melee, so vali spear would be better at maze clearing than it was before, due to not needing to hug walls to clear them.

The melee castes that the damage could be a threat to have dashes to get ontop of the person or stun them - runner, hunter, warrior namely. Vali's aren't exactly omnipotent anymore due to the aforementioned jetpack nerfs, as well diag block being added back (not saying that is a bad thing mind you) and being near a maze requires good placement spear or not due to grabs and flings. Melee castes without a gap close like Puppeteer, Gorger and Defiler have either speed buffs to get in or in Puppeteer's case is just quite fast enough to evade clicks alongside it's puppet army.

**Comment by Victor239** [comment]
I'm not a fan of this mainly because it further destroys the faction identities of xeno melee and marine guns. Marines already have high survivability in melee thanks to vali heals, tyr and jetpack, so this is a playstyle that detracts from the game rather than adds to it.

**Review by Kuro020 — CHANGES_REQUESTED** [review]
its missing the altered sprite for when chems are activated
also you modified all the other vali sprites lol

**Review by Kuro020 — APPROVED** [review]
its icondiff that is shitting itself, lgtm

**Comment by ivanmixo** [comment]
It was removed because you can kite xenos, not because of vali heals

**Comment by lKiseki** [comment]
The Vali spear was never the issue. Arguments can be made about vali healing, resource gain, and tramadol, but the weapon itself is not the culprit. Back in the day, it existed alongside Russian Red and unnerfed nanites, and I daresay it was caught in the crossfire. It never should've been removed, and instead, everything else around it should've been addressed properly instead of essentially looking at a random nearby bystander and shooting them down.

**Comment by Barnet2** [comment]
I could figure out a way to reduce slowdown on spear specifically if kiting is the issue as that is what would enable it given you are just going to be faster than a marine as a melee caste without any form of slowdown, but I didn't exactly have an issue with that when playing against spear in it's prime - moreso that it was literally unkillable due to the things around it while having the two tile range. In truth I don't know what to do about machete as it barely had a place in the game when shield was a thing.

**Comment by iforgotmeword** [comment]
i would be fine with it as long as it not just a upgrade to vail sword 

(i would like to see the spear have a wield delay so stuns are even more dangerous
or something in that sense)

**Comment by Xander3359** [comment]
So we've come full circle now huh?
Wasn't this removed because MJP declared it to be powercreep? This is just re-adding that exact powercreep back in.

**Comment by ivanmixo** [comment]
Could you please address some of the points from the PR you are reverting at least?

**Comment by Pariah919** [comment]
Why are we doing this again? Are we literally just moving in circular patterns?

Vali spear *is* an issue. My issue with vali spear is *its sole existance*. You have *not* read why I removed it.

image

I did not give a rats ass about RR, I didn't give a rats ass about nanites. I literally did not think of any of them when I made the PR to yeet vali spear to the shadowrealm where it deserves to perish forevermore as a mistake of game design that breaks the fact that vali *forces* you to be 0-1 range by allowing you to reach an extra tile and be basically immune to all xenomorph melee if you're not out of stamina. Nothing about this has been addressed in this PR. You are just attempting to readd it with justification that other godawful items have 'merely' been removed, therefore you are allowed to add this back in.

**Comment by hivemuscle** [comment]
> many xenos not being as strong as they are now
did i missed some xeno buffs?

**Comment by Henri9576** [comment]
vali was never that much of an issue in my eyes, the only time you'd die to it is when you're caught low health, the guy is using tram and he has backup to shoot at you

honestly wouldn't mind the slowdown from tram getting nerfed or just axed, vali is just a unique playstyle that i could rely on when i didn't want to use guns

it also currently sucks ass since now that shield is gone, if you try to go pure melee, you'll just get pounced by a warrior from the ether and tackled into a horde of 20 xenos, pretty much requiring you to get a gun and use vali exclusively as a side heal

tldr: you either make it op for marines by readding spear or keep it op for xenos by not readding spear, there is no inbetween

**Comment by Wisemonster** [comment]
This pr also doesn't factor in that directional attacks make it much easier for the spear to land hits,

**Comment by Tailiik** [comment]
> I could figure out a way to reduce slowdown on spear specifically if kiting is the issue as that is what would enable it given you are just going to be faster than a marine as a melee caste without any form of slowdown, but I didn't exactly have an issue with that when playing against spear in it's prime - moreso that it was literally unkillable due to the things around it (nanites which have been reworked and RR which has been removed, then pre-nerf jetpack as the cherry on top) while having the two tile range. In truth I don't know what to do about machete as it barely had a place in the game when shield was a thing.

Machete having the sword bumpslash from your offhand could be funny maybe, do some pistol gaming while having that as well

**Comment by SimplySmiley123** [comment]
> I could figure out a way to reduce slowdown on spear specifically if kiting is the issue as that is what would enable it given you are just going to be faster than a marine as a melee caste without any form of slowdown, but I didn't exactly have an issue with that when playing against spear in it's prime - moreso that it was literally unkillable due to the things around it (nanites which have been reworked and RR which has been removed, then pre-nerf jetpack as the cherry on top) while having the two tile range. In truth I don't know what to do about machete as it barely had a place in the game when shield was a thing.

Machete's only reasons to exist at the moment is to be a healing option that doesn't commit to the claymore.
Smack some xenos, get the green blood, forget about it entirely after a while, use your gun and use green blood for healing.

It's exclusively a method of healing now that the shield is gone.

**Comment by Ragecrabbo** [comment]
Would like to see some changes so that vali spear does not invalidate other forms of vali (nerfed debuffs compared to other vali weapons or maybe even slowdown when wielded since damage isnt really main reason people use vali anyway) but spear has never struck me as that opressive or annoying to fight. 
Hopefully we will see this weapon again as ive never been a fan of banishing cool things to the shadowrealm for being simply poorly implemented.

**Comment by MiniMeatwad** [comment]
Ok, but give xenos 2 tile melee range. Lol.

**Comment by 1kajowsmol** [comment]
I think a slowdown when wielded would be huge. I think removal of tram slowdown would also be something that should be on the table

**Comment by Coastfront** [comment]
Flames lose their warmth to flames; In a forest of candle-wicks, the bonfire's memory lives half as bright. 

Vali spear, was made worse by mechanics surrounding it; This is a fact.
Vali spear, was still shittily designed, regardless of what mechanics were surrounding it; This is as-well true.

And as a tool, poorly weighted is bid parting by a craftsmans hands; Ever aptly does the river fill an emptied valley,
Vali has recieved a new melee option (The claymor) that is both more-fitting towards the playstyle, and better balanced than "Vali except it doesen't get hit as often anymore."
Re-adding the spear would have it both step on the toes of its new compatriot, & this PR is made with little acknowledgement to the vali-spear still having been poorly thought out.
It wasn't solely a victim of a playstyle ear-marked by unbalanced utilities; It too was a contributor to that playstyle being poorly balanced.

**Comment by chizzy376** [comment]
> > Ok, but give xenos 2 tile melee range. Lol.
> 
> Unironically nerf rav damage and give it 2-tile melee range.

Nah just buff the damage and keep it 2 tile (not 2.5 tile like the spear is) so rav is a force to be reckoned with

**Comment by SimplySmiley123** [comment]
> > > Ok, but give xenos 2 tile melee range. Lol.
> > 
> > 
> > Unironically nerf rav damage and give it 2-tile melee range.
> 
> Nah just buff the damage and keep it 2 tile (not 2.5 tile like the spear is) so rav is a force to be reckoned with



> > > Ok, but give xenos 2 tile melee range. Lol.
> > 
> > 
> > Unironically nerf rav damage and give it 2-tile melee range.
> 
> Nah just buff the damage and keep it 2 tile (not 2.5 tile like the spear is) so rav is a force to be reckoned with

Nah.

**Comment by Henri9576** [comment]
> > > Ok, but give xenos 2 tile melee range. Lol.
> > 
> > 
> > Unironically nerf rav damage and give it 2-tile melee range.
> 
> Nah just buff the damage and keep it 2 tile (not 2.5 tile like the spear is) so rav is a force to be reckoned with

by 2.5 tile i assume you're referring to it being able to hit xenos behind walls if they're 2 tiles away from the wielder?

**Comment by ivanmixo** [comment]
There is no half tiles

**Comment by iforgotmeword** [comment]
> > > > Ok, but give xenos 2 tile melee range. Lol.
> > > 
> > > 
> > > Unironically nerf rav damage and give it 2-tile melee range.
> > 
> > 
> > Nah just buff the damage and keep it 2 tile (not 2.5 tile like the spear is) so rav is a force to be reckoned with
> 
> by 2.5 tile i assume you're referring to it being able to hit xenos behind walls if they're 2 tiles away from the wielder?

i think it means this
Untitled
2 is all of the red
2.5 is all of the blocks
image

**Comment by Henri9576** [comment]
> i think it means this Untitled 2 is all of the red 2.5 is all of the blocks image

I feel like that is just wildly inconsistent and can lead to it being very hard to adapt to using the spear
However, the thing I see most xenos complain about is this scenario where you can still hit xenos even if they're behind a wall, so there should likely be a solution for that
vali

**Comment by Henri9576** [comment]
> There is no half tiles

They did not mean it literally, although I'm still trying to figure out what exactly they meant

**Comment by SimplySmiley123** [comment]
> > i think it means this Untitled 2 is all of the red 2.5 is all of the blocks image
> 
> I feel like that is just wildly inconsistent if it were to be added and can lead to it being very hard to adapt to using the spear However, the thing I see most xenos complain about is this scenario where you can still hit xenos even if they're behind a wall, so there should likely be a solution for that vali

This isn't even that consistent to do as a marine with regular spear.
Sometimes it will hit something "behind" the wall, sometimes it refuses entirely.

**Comment by Zangeiti** [comment]
> Its not like its super OP or anything, its just 'existing item +1' What is the point of vali sword if vali spear is in the game? By that metric it's powercreep.

vali sword is supposed to be a gimmick build where you have a freehand to use sidearms or other 1handed things

**Comment by SimplySmiley123** [comment]
> Vali spear is stinky because its just a direct upgrade to vali sword. It has the same damage, has a faster attack speed, and without shields there is no benefit to the sword's one handed use outside of getting delimbed. This isn't even bringing in the 2 tile range advantage.
> 
> Its not like its super OP or anything, its just 'existing item +1' What is the point of vali sword if vali spear is in the game? By that metric it's powercreep.

Honestly the sword does need it's own changes, even without the vali spear, _why would you ever use it anyway?_

**Comment by SimplySmiley123** [comment]
> > Its not like its super OP or anything, its just 'existing item +1' What is the point of vali sword if vali spear is in the game? By that metric it's powercreep.
> 
> vali sword is supposed to be a gimmick build where you have a freehand to use sidearms or other 1handed things

It's really, really not good at it however, mostly due to the swing delay affecting the ability to use your other hand at all. Add the chem reload being manual on the machete instead of automatic like with claymore and that it _will_ cancel the reload if you switch hands and yeahhh it provides no reasons to bother with it.

At best machete's used at the start of the op to gather green blood, then discarded later on with the marine just playing normally with a gun using green blood as healing from time to time.

**Comment by Henri9576** [comment]
> Vali spear is stinky because its just a direct upgrade to vali sword. It has the same damage, has a faster attack speed, and without shields there is no benefit to the sword's one handed use outside of getting delimbed. This isn't even bringing in the 2 tile range advantage.
> 
> Its not like its super OP or anything, its just 'existing item +1' What is the point of vali sword if vali spear is in the game? By that metric it's powercreep.

The way I see it, you're trading a free hand slot in exchange for a +1 tile range upgrade, also, the fact that when you get stunned you unwield the spear and need to wield it again after getting back up is another disadvantage, it sounds easy to do but in the middle of getting flinged around by xenos, it gets alot worse

**Comment by Wisemonster** [comment]
> The way I see it, you're trading a free hand slot in exchange for a +1 tile range upgrade, which admittedly is a much better deal with shields gone, also, the fact that when you get stunned you unwield the spear and need to wield it again after getting back up is another disadvantage, it sounds easy to do but in the middle of getting flinged around by xenos, it gets alot worse

One thing to note about the spear: You do not have to wield it in order to get the extra range. You can 1 hand it and it'll still have it's 2-tile range, meaning if you are using the spear for it's utility (such as it's slowdown or shatter) it is just better than the sword due to being easier to hit xenos with. Not to mention, wielding is instant, so the downside of dropping the wield is needing to click a button when you get up.

**Comment by Zangeiti** [comment]
> One thing to note about the spear: You do not have to wield it in order to get the extra range. You can 1 hand it and it'll still have it's 2-tile range, meaning if you are using the spear for it's utility (such as it's slowdown or shatter) it is just better than the sword due to being easier to hit xenos with. Not to mention, wielding is instant, so the downside of dropping the wield is needing to click a button when you get up.

Don't forget that the vali sword fits on the belt slot with its scabbard, unlike the spear which takes up the back or suit slot. You have more room for extra ammo/meds/guns

**Comment by Henri9576** [comment]
> One thing to note about the spear: You do not have to wield it in order to get the extra range. You can 1 hand it and it'll still have it's 2-tile range, meaning if you are using the spear for it's utility (such as it's slowdown or shatter) it is just better than the sword due to being easier to hit xenos with. Not to mention, wielding is instant, so the downside of dropping the wield is needing to click a button when you get up.

The damage is severely reduced if you one hand it, like more than half I think, sword is definetely better if you want to one hand

**Comment by SimplySmiley123** [comment]
> > One thing to note about the spear: You do not have to wield it in order to get the extra range. You can 1 hand it and it'll still have it's 2-tile range, meaning if you are using the spear for it's utility (such as it's slowdown or shatter) it is just better than the sword due to being easier to hit xenos with. Not to mention, wielding is instant, so the downside of dropping the wield is needing to click a button when you get up.
> 
> The damage is severely reduced if you one hand it, like more than half I think, sword is definetely better if you want to one hand

When one handed spear deals damage comparable to a boot knife.

**Comment by Fawcksy** [comment]
Finally, a based PR

**Comment by Kuro020** [comment]
After the xeno buffs, I dont think this is that bad anymore

**Comment by Looking4bros** [comment]
> vali spear is a direct upgrade to the sword
Vali blade is like a sidearm, it's only really useful if you can dual wield it, or your loadout permits having a sidearm for some reason. The only real upside to the sword is was recently removed, that being full damage when striking from the ground, which is niche anyway. Also, old spear had 1.2 second attacks like the blade did and does, i think this iteration is taking from the "obj/item" parent which has a 1.1 second attack, probably an oversight. Bumpslash isn't to be overlooked.

> Reducing slowdown on spear
I mean, as mentioned by others, it's only really useful as a punishment for xenos who're poorly positioned, forcing them to stay in it for 1 second longer, it used to be 2 seconds, like the charge time, but that was even before myself learning vali. I don't care for slowdown myself, only used it for damage before the update.

The spear is without a doubt the best skirmishers weapon of the four vali has, but that's boring and lame. Vali only has to be fast enough to backstep or sidestep special attacks, which is already a guessing game. And the theoretical damage output is only superior to the claymore if you can land at least twice as many hits as the big blade. And only if you use tricordrazine, or only go for lower armored xenos(less than 40 melee armor i think). Spear has the best potential for utilizing the armor reduction tricordrazine has than the other weapons, but i feel that the claymore fits my playstyle better.

The biggest buff i can think for machette, besides that crazy offhand firing pr, would be dual wielding the vali swords, both being connected, as then you could chain two charged hits one after the other, but it'd only be a buff to the people who have extremely reckless playstyles.
